README.md

```diff
@@ -1,123 +1,323 @@
+| `numberOfWorkerHarnessThreads`   | (Optional) The number of threads per each worker harness process                                                                                                                                                                                                   | All                 |
+| `dlpApiRetryCount`               | (Optional) Number of retries in case of transient errors in DLP API, Default value is 10                                                                                                                                                                           | All                 |
+| `getInitialBackoff`              | (Optional) Initial backoff (in seconds) for retries with exponential backoff, default is 5s                                                                                                                                                                        | All                 |
+
+### Supported File Formats
+
+1. CSV
+
+The sample commands for processing csv files have been provided in the above section [Build and Run](#run-the-samples)
+
+2. TSV
+
+TSV files are handled in the same way as CSV files with TAB as column delimiter. No additional changes are required in pipeline options.
+3. JSONL
+
+The pipeline supports JSONL file format where each line is a valid JSON Object and newline character is used to separate JSON objects. A sample file can be found in [test resources](src/test/resources/CCRecords_sample.jsonl). 
+To run the pipeline for JSONL files, the list of comma separated headers also needs to be passed in the pipeline options. 
 ```
 // Copy the sample jsonl file to GCS
 gsutil cp ./src/test/resources/CCRecords_sample.jsonl gs://<bucket>/
 
 // Run the pipeline using following command
 gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs=" --region=<region> --project=<projct_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=gs://<path>.jsonl --dataset=<name>   --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --DLPMethod=DEID --headers=<comma_separated_list_of_headers>"
-
 ```
+4. Avro
+
+Avro files are handled in the same way as CSV files. No additional changes are required to run the pipeline.
+
+Refer [here](src/test/resources/avro) for sample data.
 
-## ReIdentification From BigQuery 
-You can. use the pipeline to read from BgQuery table and publish the re-identification data in a secure pub sub topic.
+5. CSV files with custom delimiter 
 
-Export the Standard SQL Query to read data from bigQuery
-One example from our solution guide:
+It is possible to provide csv files with custom delimiter. The delimiter has to be passed in the pipeline option as "--columnDelimiter". 
```

Comment:
can the delimiter can be anything/any character(including UTF & ASCII)?

Reply:
Found out together that it can be anything but by default pipeline will error out parsing command if "," & "\n" is provided hence @dup05 to call that out in pipeline options

Reply:
When we say it can be anything do we mean any single "character" or can it be multi-characters or strings?

Reply:
@imadethis With https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification/pull/100 CSV with custom delimiter is supported. It can be a single ASCII char AFAIK.